test/functional/wallet_txn_doublespend.py

```diff
@@ -34,6 +34,14 @@ def setup_network(self):
     def run_test(self):
         # All nodes should start with 1,250 BTC:
         starting_balance = 1250
+
+        # All nodes should be out of IBD.
+        # If the nodes are not all out of IBD, that can interfere with
+        # blockchain sync later in the test when nodes are connected, due to
+        # timing issues.
+        for n in self.nodes:
+            assert n.getblockchaininfo()["initialblockdownload"] == False
```

Comment:
This assertion would fail when the tests are called "manually", since the cache is not reset when calling tests individually (not through) the test_runner script.

```
$ ./test/functional/wallet_txn_doublespend.py 
2019-02-08T17:09:47.027000Z TestFramework (INFO): Initializing test directory /tmp/bitcoin_func_test_62_g4dwe
2019-02-08T17:09:48.015000Z TestFramework (INFO): Stopping nodes
2019-02-08T17:09:48.275000Z TestFramework (INFO): Cleaning up /tmp/bitcoin_func_test_62_g4dwe on exit
2019-02-08T17:09:48.275000Z TestFramework (INFO): Tests successful

[... time travel to +1 day]

$ ./test/functional/wallet_txn_doublespend.py 
2019-02-09T17:10:07.360000Z TestFramework (INFO): Initializing test directory /tmp/bitcoin_func_test_jck9m01h
2019-02-09T17:10:08.138000Z TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "/home/marco/workspace/bitcoin/test/functional/test_framework/test_framework.py", line 176, in main
    self.run_test()
  File "./test/functional/wallet_txn_doublespend.py", line 43, in run_test
    assert n.getblockchaininfo()["initialblockdownload"] == False
AssertionError

Reply:
Ah, I misunderstood how the cache worked (I thought it only persisted between calls to tests in the same test_runner run, and otherwise was being cleared between tests).